### PR TITLE
Resource Health RP changes for new resource type /events and /impacte…

### PR DESCRIPTION
**Pull request by HimanshuMisra1**
Resource Health RP changes for new resource type /events and /impactedResources.

This is a re-open from previous PR https://github.com/Azure/azure-rest-api-specs/pull/2127 which was closed due to inactivity around an year back. Starting a new one with first iteration for changes which were already signed off by ARM (please check label in previous PR). Will add another update/iteration with changes that have happened since then, should be minor additions of additional properties.

We want a new preview api-version 2018-01-01-preview.

### CODE REVIEW

**Review by sergey-shandar — APPROVED** [review]

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2018-08-01/ResourceHealth.json [hunk 1]
```diff
@@ -0,0 +1,1050 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ResourceHealth/events": {
```
Comment (KrisBash) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2018-08-01/ResourceHealth.json]: For scope extension, you will need additional paths (RG and resourceUri).

**Review by KrisBash — APPROVED** [review]
Please see comment regarding missing Extension resource paths

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2018-08-01/ResourceHealth.json (hunk 1 above)
Comment (HimanshuMisra1) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2018-08-01/ResourceHealth.json]: Added additional path for resourceUri. We will expose for RG scope later.

**Comment by sergey-shandar** [comment]
@HimanshuMisra1 could you fix model validation errors https://travis-ci.org/Azure/azure-rest-api-specs/jobs/464693921 ?

**Comment by anuchandy** [comment]
There seems couple of model Validation errors still pending https://travis-ci.org/Azure/azure-rest-api-specs/jobs/466782277, please take a look.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**